lib/rom/sql/plugin/pagination.rb

```diff
@@ -31,7 +31,9 @@ def total
           end
 
           def total_pages
-            (total / per_page) + 1
+            total_pages = total / per_page
+            total_pages += 1 unless total % per_page == 0
+            total_pages
```

Comment:
Nice one!

For further reference, Sequel's [pagination module](https://github.com/jeremyevans/sequel/blob/master/lib/sequel/extensions/pagination.rb) considers that the dataset would have 1 page when there are no elements at all.

```ruby
@page_count = (record_count / page_size.to_f).ceil
@page_count = 1 if @page_count == 0
```

I haven't. I'll leave to you such decision